[doc][tune] Add tune checkpoint user guide. (#33145)
ray-project/ray: doc/source/tune/tutorials/tune-distributed.rst

        scheduling:
          - preemptible: true

> Spot instances may be removed suddenly while trials are still running. Often times this may be difficult to deal with when using other distributed hyperparameter optimization frameworks. Tune allows users to mitigate the effects of this by preserving the progress of your model training through :ref:`checkpointing <tune-trial-checkpoint>`.

Author: Actually removed the sentence "This can cause problems when using ..."

Feels like we are saying other non-ray-tune choices are not good, without being really specific about it.

> Tune will automatically restart trials in case of trial failures/error (if ``max_failures != 0``), both in the single node and distributed setting.

Reviewer: ```suggestion
Tune automatically restarts trials in the case of trial failures or errors (if ``max_failures != 0``), both in the single node and distributed setting.
```

> Tune will restore trials from the latest checkpoint, where available. In the distributed setting, Tune will automatically sync the trial folder with the driver. For example, if a node is lost while a trial (specifically, the corresponding Trainable actor of the trial) is still executing on that node and a checkpoint of the trial exists, Tune will wait until available resources are available to begin executing the trial again.

Reviewer: ```suggestion
Tune restores trials from the latest available checkpoint. In the distributed setting, Tune automatically syncs the trial folder with the driver. For example, if a node is lost while a trial (specifically, the corresponding Trainable actor of the trial) is still executing on that node and a checkpoint of the trial exists, Tune waits until available resources are available to begin executing the trial again.
```

See :ref:`here for information on checkpointing <tune-trial-checkpoint>`.

> If the trial/actor is placed on a different node, Tune will automatically push the previous checkpoint file to that node and restore the remote trial actor state, allowing the trial to resume from the latest checkpoint even after failure.

Reviewer: ```suggestion
If the trial or actor moves to a different node, Tune automatically pushes the previous checkpoint file to that node and restores the remote trial actor state, allowing the trial to resume from the latest checkpoint even after failure.
```

Author: changed to "is then placed to..."
